### PR TITLE
Add CRUDs for missing fields for API Hosts entity

### PR DESCRIPTION
**Pull request by svtkachenko**
Part of #3270, depends on SatelliteQE/nailgun#346 and SatelliteQE/nailgun#347.
`test_positive_update_provision_method` was not added as "Provision method can't be updated after host is provisioned".

```python
% pytest -v tests/foreman/api/test_host.py -k 'test_positive_create_with_puppet_class or test_positive_create_with_content_view or test_positive_create_with_host_parameters or test_positive_create_with_image or test_positive_create_with_provision_method or test_positive_update_puppet_class or test_positive_update_content_view or test_positive_update_host_parameters or test_positive_update_image or test_positive_update_provision_method'
===================================================================== test session starts =====================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.4, py-1.4.31, pluggy-0.4.0 -- /home/qui/code/venv/2/bin/python2
cachedir: .cache
rootdir: /home/qui/code/robottelo, inifile: 
plugins: xdist-1.15.0, html-1.12.0, cov-2.4.0
collected 60 items 

tests/foreman/api/test_host.py::HostTestCase::test_positive_create_with_content_view <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_host.py::HostTestCase::test_positive_create_with_host_parameters <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_host.py::HostTestCase::test_positive_create_with_image <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_host.py::HostTestCase::test_positive_create_with_provision_method <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_host.py::HostTestCase::test_positive_create_with_puppet_class <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_host.py::HostTestCase::test_positive_update_content_view <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_host.py::HostTestCase::test_positive_update_host_parameters <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_host.py::HostTestCase::test_positive_update_image <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_host.py::HostTestCase::test_positive_update_puppet_class <- robottelo/decorators/__init__.py PASSED

===================================================================== 51 tests deselected =====================================================================
========================================================== 9 passed, 51 deselected in 177.73 seconds ==========================================================
```

### CODE REVIEW

File: tests/foreman/api/test_host.py [hunk 1]
```diff
@@ -37,12 +41,42 @@
...
+        cls.cv = publish_puppet_module(
+            [{'author': 'robottelo', 'name': 'generic_1'}],
+            CUSTOM_PUPPET_REPO,
+            organization_id=cls.org.id
+        )
+        cls.env = entities.Environment().search(
+            query={'search': u'content_view="{0}"'.format(cls.cv.name)}
+        )[0].read()
+        cls.lce = entities.LifecycleEnvironment().search(query={
+            'search': 'name=Library and organization_id={0}'.format(
```
Comment (oshtaier) [inline tests/foreman/api/test_host.py]: `name=ENVIRONMENT`

File: tests/foreman/api/test_host.py [hunk 2]
```diff
@@ -383,6 +437,95 @@ def test_positive_create_with_compute_profile(self):
         self.assertEqual(host.compute_profile.read().name, profile.name)
 
     @run_only_on('sat')
+    @tier2
+    def test_positive_create_with_content_view(self):
```
Comment (oshtaier) [inline tests/foreman/api/test_host.py]: Basically, covered in https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/api/test_contentview.py#L94
But probably, that one is more basic

**Comment by oshtaier** [comment]
ACK pending nitpicks

File: tests/foreman/api/test_host.py [hunk 3]
```diff
@@ -22,7 +22,11 @@
 from fauxfactory import gen_integer, gen_ipaddr, gen_mac, gen_string
 from nailgun import client, entities
 from requests.exceptions import HTTPError
+from six.moves import http_client
```
Comment (svtkachenko) [inline tests/foreman/api/test_host.py]: Isn't `six` a third-party module?
And according to http://robottelo.readthedocs.io/en/latest/code_standards.html#recommended
`third-party` should be placed before `application-specific` and separated with `a blank line between groups of imports`.

Comment (renzon) [inline tests/foreman/api/test_host.py]: I agree with you and most of times I refactored my code to put them all together. I knew [pep8](https://www.python.org/dev/peps/pep-0008/#imports) state about importing but never realized robottello doc itself mention the exactly same format. Maybe it's worth a email or issue discussion about it so everyone get aware of this. I would be happier once Pycharm follow pep 8 as well ;)

**Pull request merged**